icepyx/core/hdf5_arrow_table.py

```diff
@@ -0,0 +1,181 @@
+        """Read datasets from the HDF5 file using h5coro."""
+        variable_paths = self.datasets.variables
+        dataset_objs = [dict(v) for k, v in variable_paths.items()] + [dict(self.datasets.longitude), dict(self.datasets.latitude)]
+        return self.h5obj.readDatasets(dataset_objs, block=False)
+
+    def _create_mask(self) -> np.ndarray:
+        """Create a mask for the datasets if masking is requested."""
+        if not self.mask_using:
+            return None
+        
+        mask_dataset_path = self.datasets.variables[self.mask_using].dataset
+        mask_dataset_points = self.promise[mask_dataset_path]
+        attributes = self.h5obj.list(mask_dataset_path)[1]
+        fill_value = attributes['_FillValue']
+        mask = (mask_dataset_points != fill_value)
+        
+        return mask if mask.any() else None
+
+    def _process_datasets(self, mask: np.ndarray):
+        """Process each dataset, apply mask if necessary, and create pyarrow arrays and schema fields."""
+        schema_fields = []
+        pyarrow_arrays = {}
+
+        for k, v in self.datasets.variables.items():
+            if k == self.mask_using:
+                continue
+            
+            variable_points = self.promise[v.dataset]
+            if mask is not None:
+                variable_points = variable_points[mask]
+            
+            schema_fields.append(pa.field(k, pa.from_numpy_dtype(variable_points.dtype)))
+            pyarrow_arrays[k] = pa.array(variable_points)
+
+        if self.mask_using:
+            mask_dataset_path = self.datasets.variables[self.mask_using].dataset
+            mask_dataset_points = self.promise[mask_dataset_path][mask]
+            schema_fields.append(pa.field(self.mask_using, pa.from_numpy_dtype(mask_dataset_points.dtype)))
+            pyarrow_arrays[self.mask_using] = pa.array(mask_dataset_points)
+
+        return pyarrow_arrays, schema_fields
+
+    # Question: Does it make sense to add other geometry options, like polygon, line, etc.
+    def _create_geometry_column(self, mask: np.ndarray) -> pa.FixedSizeListArray:
+        """Create a geometry column from latitude and longitude."""
+        latitude = self.promise[self.datasets.latitude.dataset]
+        longitude = self.promise[self.datasets.longitude.dataset]
+        
+        if mask is not None:
+            latitude = latitude[mask]
+            longitude = longitude[mask]
+
+        np_coords = np.column_stack([longitude, latitude])
+        flat_coords = pa.array(np_coords.flatten("C"))
+        return pa.FixedSizeListArray.from_arrays(flat_coords, 2)
+
+    def _build_table(self, pyarrow_arrays: dict, schema_fields: list, geom_points: pa.FixedSizeListArray) -> pa.Table:
+        """Construct the final pyarrow Table with the geometry column."""
+        geom_field = pa.field("geometry", geom_points.type, metadata={b'ARROW:extension:name': b'geoarrow.point'})
+        schema_fields.append(geom_field)
```

Comment:
Does Icesat 2 have a specific CRS? You should add the CRS in the metadata as well:

```py
import json

ext_meta = json.dumps({"crs": {projjson obj}})
metadata[b"ARROW:extension:metadata"] = ext_meta
```